app/scripts/services/deployments.js

```diff
@@ -401,6 +401,45 @@ angular.module("openshiftConsole")
       return byDC;
     };
 
+    // Sorts replication controllers that are part of a deployment config by
+    // the deployment version annotation.
+    DeploymentsService.prototype.sortByDeploymentVersion = function(replicationControllers, descending) {
+      var compareDeployments = function(left, right) {
+        var leftVersion = annotation(left, 'deploymentVersion');
+        var rightVersion = annotation(right, 'deploymentVersion');
+
+        // Fall back to sorting by name if no deployment versions.
+        var leftName, rightName;
+        if (!leftVersion && !rightVersion) {
+          leftName = _.get(left, 'metadata.name', '');
+          rightName = _.get(right, 'metadata.name', '');
+          if (descending) {
+            return rightName.localeCompare(leftName);
+          }
+          return leftName.localeCompare(rightName);
+        }
+
+        if (!leftVersion) {
+          return descending ? 1 : -1;
+        }
+
+        if (!rightVersion) {
+          return descending ? -1 : 1;
+        }
+
+        leftVersion = parseInt(leftVersion, 10);
+        rightVersion = parseInt(rightVersion, 10);
```

Comment:
shouldn't we handle bad version annotations that parse as NaN?  would expect these to be treated like non-existing versions